hera_sim/simulate.py

```diff
@@ -61,39 +76,61 @@ def __init__(self, data=None, defaults_config=None, **kwargs):
         self._initialize_data(data, **kwargs)
 
     @property
-    def antpos(self):
-        # TODO: docstring
-        """
-        """
+    def antpos(self) -> Dict[int, Tuple[float, float, float]]:
+        """Dictionary of antenna numbers mapped to ENU positions."""
         antpos, ants = self.data.get_ENU_antpos(pick_data_ants=True)
         return dict(zip(ants, antpos))
 
     @property
-    def lsts(self):
-        # TODO: docstring
+    def lsts(self) -> np.ndarray:
+        """Unique LSTs in the data array."""
         return np.unique(self.data.lst_array)
 
     @property
-    def freqs(self):
+    def freqs(self) -> np.ndarray:
         """Frequencies in GHz."""
         return np.unique(self.data.freq_array) / 1e9
 
     @property
-    def times(self):
-        """Return unique simulation times."""
+    def times(self) -> np.ndarray:
+        """Unique simulation JDs."""
         return np.unique(self.data.time_array)
 
     def apply_defaults(self, config, refresh=True):
-        # TODO: docstring
-        """
-        """
+        """Apply a given set of defaults."""
         # actually apply the default settings
+        # TODO: SGM: why is this a method?
         defaults.set(config, refresh=refresh)
 
-    def add(self, component, **kwargs):
-        # TODO: docstring
-        """
-        """
+    def add(
+        self, component: Union[Type[SimulationComponent], str], **kwargs
+    ) -> Optional[UVData]:
+        """Add a particular simulation component to the simulated data.
+
+        Parameters
+        ----------
+        component
+            Either a string name (or alias) of a component model to add,
+            or its actual class.
```

Comment:
It can also be an instance of the class.

Reply:
Good to know. Fixed.